app/src/main-process/app-window.ts

```diff
@@ -317,16 +317,23 @@ export class AppWindow {
     this.window.destroy()
   }
 
+  /**
+   * Method to show the save dialog and return the first file path it returns.
+   */
+  public async showSaveDialog(options: Electron.SaveDialogOptions) {
+    const { canceled, filePath } = await dialog.showSaveDialog(
+      this.window,
+      options
+    )
+
+    return !canceled ? filePath ?? null : null
+  }
+
   /**
    * Method to show the open dialog and return the first file path it returns.
    */
   public async showOpenDialog(options: Electron.OpenDialogOptions) {
     const { filePaths } = await dialog.showOpenDialog(this.window, options)
-
-    if (filePaths.length === 0) {
-      return null
-    }
-
-    return filePaths[0]
+    return filePaths[0] ?? null
```

Comment:
I know I suggested this approach but it _just_ dawned on me that since TypeScript doesn't think that filePath[0] can ever be undefined the return type of `showOpenDialog` will be `Promise<string>` instead of `Promise<string | null>`.

```suggestion
    return filePaths.length > 0 ? filePaths[0] : null
```